### PR TITLE
chore: move `/gitauth` to `/externalauth` on the frontend

### PR DESCRIPTION
**Pull request by kylecarbs**
This actually took a lot more jank than anticipated, so I wanted to split this up before adding the ability to embed new providers.

### CODE REVIEW

**Comment by Emyrk** [comment]
Is the goal to add more oauth2 providers?

**Comment by kylecarbs** [comment]
@Emyrk the goal is to enable any OAuth2 or OIDC provider to become a "Click to Login" during workspace creation.

**Review by Emyrk — APPROVED** [review]
Pretty sure I commented on the only actual Go code changes that are not just a rename refactor. If that is all this PR is (A rename refactor), then green check the Go code.

I have not looked at all the existing tests for external auth, but if we are going to refactor/add/modify them, lets use the `oidctest` package and do it right.

File: coderd/coderd.go [hunk 1]
```diff
@@ -546,21 +546,24 @@ func New(options *Options) *API {
...
-				continue
+	// We must support gitauth and externalauth for backwards compatibility.
+	for _, route := range []string{"gitauth", "externalauth"} {
+		r.Route("/"+route, func(r chi.Router) {
+			for _, externalAuthConfig := range options.ExternalAuthConfigs {
+				// We don't need to register a callback handler for device auth.
+				if externalAuthConfig.DeviceAuth != nil {
+					continue
+				}
+				r.Route(fmt.Sprintf("/%s/callback", externalAuthConfig.ID), func(r chi.Router) {
```
[the remark this answers is not in the document]
Comment (kylecarbs) [reply coderd/coderd.go:557]: Nah, gitauth supports all git types, so I think it's better to just allow both for now.

Maybe we log a warn for the gitauth reqs or something

File: cli/create_test.go [hunk 2]
```diff
@@ -628,7 +628,7 @@ func TestCreateWithGitAuth(t *testing.T) {
 	clitest.Start(t, inv)
 
 	pty.ExpectMatch("You must authenticate with GitHub to create a workspace")
-	resp := coderdtest.RequestGitAuthCallback(t, "github", client)
+	resp := coderdtest.RequestExternalAuthCallback(t, "github", client)
```
[the remark this answers is not in the document]
Comment (kylecarbs) [reply cli/create_test.go:631]: Yeah, we def should. When I make the actual changes I will look into this

**Pull request merged**